libs/execution/src/lib/constraints/executors/range-constraint-executor.spec.ts

```diff
@@ -52,10 +53,15 @@ describe('Validation of RangeConstraintExecutor', () => {
       document.parseResult.value,
       'pipelines@0/blocks@2',
     ) as BlockDefinition;
-    const constraint = locator.getAstNode<TypedConstraintDefinition>(
-      document.parseResult.value,
-      'constraints@0',
-    ) as TypedConstraintDefinition;
+
+    const allElements = AstUtils.streamAllContents(document.parseResult.value);
```

Comment:
Can be merged like this but the code is repeated a lot. Much better would be to write a `expectConstraintsToExist` util method, put it into a test util file and just call it everywhere.

Reply:
On the one hand, I agree that the code repeats quite a bit here. On the other hand, it is test-case specific test setup. Generalizing here might introduce coupling between tests...

So I ended up creating a helper function that is flexible by passing a customizable filter function as parameter.